api/control/v1alpha1/types.go

```diff
@@ -142,6 +136,18 @@ const (
 	BindingPolicyConditionMisconfigured string = "BindingPolicyMisconfigured"
 )
 
+// DownsyncObjectTestAndStatusCollection identifies some objects (by a predicate)
+// and asks for some combined status to be returned from those objects.
+// The latter is dictated through applying a set of StatusCollectors,
+type DownsyncObjectTestAndStatusCollection struct {
+	DownsyncObjectTest `json:",inline"`
+	// statusCollectors is a list of StatusCollectors name references that are applied to the selected objects.
+	StatusCollectors []string `json:"statusCollectors,omitempty"`
+	// inlinedStatusCollectors is a list of StatusCollectorSpec that are inlined in the BindingPolicy.
+	// The StatusCollectors in this list are applied to the selected objects.
+	// InlinedStatusCollectors   []StatusCollectorSpec `json:"inlinedStatusCollectors,omitempty"` TODO
```

Comment:
If we are going to go with the distinct API objects then I would not recommend adding the complexity of also having inlining.

Reply:
This can be pushed until it is clearly needed, if ever.